src/app/shared/components/addregressionregion/addregressionregion.component.ts

```diff
@@ -0,0 +1,422 @@
+    });
+    const tileLayer_streets = L.tileLayer('https://server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer/tile/{z}/{y}/{x}', {
+      attribution: 'Tiles &copy; Esri &mdash; Source: Esri, DeLorme, NAVTEQ, USGS, Intermap, iPC, NRCAN, Esri Japan, METI, Esri China (Hong Kong), Esri (Thailand), TomTom, 2012'
+    });
+    const tileLayer_grayscale = L.tileLayer('https://server.arcgisonline.com/ArcGIS/rest/services/Canvas/World_Light_Gray_Base/MapServer/tile/{z}/{y}/{x}', {
+      attribution: 'Tiles &copy; Esri &mdash; Esri, DeLorme, NAVTEQ'
+    });
+    const tileLayer_satellite = L.tileLayer('https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}', {
+        attribution: 'Tiles &copy; Esri &mdash; Source: Esri, i-cubed, USDA, USGS, AEX, GeoEye, Getmapping, Aerogrid, IGN, IGP, UPR-EGP, and the GIS User Community'
+    });
+
+    // Initialize map 
+    this.map = new L.Map('map', {
+      center: new L.LatLng(39.8283, -98.5795),
+      zoom: 4,
+      layers: [tileLayer_streets]
+    });
+
+    // Create basemap collection
+    const baseMaps = {
+      'Streets': tileLayer_streets,
+      'Topography': tileLayer_topography,
+      'Grayscale': tileLayer_grayscale,
+      'Satellite': tileLayer_satellite
+    };
+
+    // Add basemap control to the map
+    L.control.layers(baseMaps).addTo(this.map);
+
+  }
+
+  public getRegRegions() {
+    // moving to own function for when new regression region is added
+    this._settingsService.getEntities(this.configSettings.regionURL + this.selectedRegion.id + '/' + this.configSettings.regRegionURL)
+      .subscribe(res => {
+        if (res.length > 1) { res.sort((a, b) => a.name.localeCompare(b.name)); }
+        this.regressionRegions = res;
+      });
+  }
+
+  public showModal(): void {
+    if (this.selectedRegion) { this.getRegRegions(); }
+    this.modalRef = this._modalService.open(this.modalElement, { backdrop: 'static', keyboard: false, size: 'lg' });
+    this.modalRef.result.then(
+      result => {
+        // this is the solution for the first modal losing scrollability
+        if (document.querySelector('body > .modal')) {
+          document.body.classList.add('modal-open');
+        }
+        this.CloseResult = `Closed with: ${result}`;
+      },
+      reason => {
+        this.CloseResult = `Dismissed ${this.getDismissReason(reason)}`;
+        this.cancelCreateRegression();
+      }
+    );
+  }
+
+  outputWimMessages(msg) {
+    // output messages from http request to toast
+    for (const key of Object.keys(msg)) {
+      for (const item of msg[key]) {
+        this._toasterService.pop(key, key.charAt(0).toUpperCase() + key.slice(1), item);
+      }
+    }
+  }
+
+  public showAddRegRegion() {
+    const addRegRegForm: AddRegressionRegion = {
+      show: true,
+      regRegionID: null
+  }
+    this._nssService.setAddRegressionRegionModal(addRegRegForm);
+  }
```

Comment:
This looks exactly like the call from in the mainview component. I don't know why it's in this component. I would assume we'd use it to invoke this component from another component. It might be another function that was carried over from the add scenario component by accident.